src/agents/auto_fan.py

```
@@ -0,0 +1,106 @@
+def _mock_fan_message(
+    voice: FanVoice, history: list[dict[str, str]], spend_meta: dict[str, Any] | None
+) -> str:
+    last_assistant = ""
+    for msg in reversed(history):
+        if msg.get("role") == "assistant":
+            last_assistant = str(msg.get("content") or "").strip()
+            break
+    sig = _mock_voice_signal(voice, spend_meta)
+    if last_assistant:
+        return f"[{sig}] haha you're cute 😅 {last_assistant[:120]}"
+    return f"[{sig}] hey! just checking in — how’s your day going?"
```

Ruff (0.15.12)

[warning] String contains ambiguous `’` (RIGHT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF001)